File: packages/google_mlkit_genai_speech_recognition/ios/Classes/GoogleMlKitGenaiSpeechRecognitionPlugin.m
#import <Foundation/Foundation.h>
#import "GoogleMlKitGenaiSpeechRecognitionPlugin.h"

#define channelName @"google_mlkit_genai_speech_recognition"
#define checkStatus @"genai#checkStatus"
#define startRecognition @"genai#startRecognition"
#define stopRecognition @"genai#stopRecognition"
#define closeSpeechRecognizer @"genai#closeSpeechRecognizer"

@implementation GoogleMlKitGenaiSpeechRecognitionPlugin {
    NSMutableDictionary *instances;
}

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:channelName
                                     binaryMessenger:[registrar messenger]];
    GoogleMlKitGenaiSpeechRecognitionPlugin* instance = [[GoogleMlKitGenaiSpeechRecognitionPlugin alloc] init];
    [registrar addMethodCallDelegate:instance channel:channel];
}

- (id)init {
    self = [super init];
    if (self)
        instances = [NSMutableDictionary dictionary];
    return self;
}

- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
    if ([call.method isEqualToString:checkStatus]) {
        // iOS implementation would go here
        // Note: GenAI APIs are currently Android-only
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:startRecognition]) {
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:stopRecognition]) {
        result([FlutterError errorWithCode:@"UNIMPLEMENTED"
                                   message:@"GenAI APIs are currently only available on Android"
                                   details:nil]);
    } else if ([call.method isEqualToString:closeSpeechRecognizer]) {
        NSString *uid = call.arguments[@"id"];
        [instances removeObjectForKey:uid];
        result(NULL);
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end

File: packages/google_mlkit_genai_speech_recognition/ios/google_mlkit_genai_speech_recognition/Sources/google_mlkit_genai_speech_recognition/GoogleMlKitGenaiSpeechRecognitionPlugin.swift
import Flutter

@objc
public class GoogleMlKitGenaiSpeechRecognitionPlugin: NSObject, FlutterPlugin {
  private var instances: [String: Any] = [:]

  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "google_mlkit_genai_speech_recognition",
      binaryMessenger: registrar.messenger()
    )
    let instance = GoogleMlKitGenaiSpeechRecognitionPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    let unimplemented = FlutterError(
      code: "UNIMPLEMENTED",
      message: "GenAI APIs are currently only available on Android",
      details: nil
    )
    switch call.method {
    case "genai#checkStatus", "genai#startRecognition", "genai#stopRecognition":
      result(unimplemented)
    case "genai#closeSpeechRecognizer":
      if let args = call.arguments as? [String: Any], let uid = args["id"] as? String {
        instances.removeValue(forKey: uid)
      }
      result(nil)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
